cli/command/swarm/join.go

```diff
@@ -50,6 +53,15 @@ func runJoin(dockerCli *command.DockerCli, opts joinOptions) error {
 		AdvertiseAddr: opts.advertiseAddr,
 		RemoteAddrs:   []string{opts.remote},
 	}
+	availability := strings.ToLower(opts.availability)
```

Comment:
It might make sense to do the `swarm.NodeAvailability` cast here, to avoid the casts below.